src/compiler/program.ts

```diff
@@ -3882,14 +3887,19 @@ namespace ts {
         }
     }
 
-    function getModuleNames({ imports, moduleAugmentations }: SourceFile): string[] {
+    function getModuleNames(file: SourceFile, options: CompilerOptions): string[] {
+        const { imports, moduleAugmentations } = file;
         const res = imports.map(i => i.text);
         for (const aug of moduleAugmentations) {
             if (aug.kind === SyntaxKind.StringLiteral) {
                 res.push(aug.text);
             }
             // Do nothing if it's an Identifier; we don't need to do module resolution for `declare global`.
         }
+        const jsxRuntimeImport = getJSXRuntimeImport(getJSXImplicitImportBase(options, file), options);

```

Comment:
You need test for incremental build and need to be handled in builder to store the file reference with this import.